### PR TITLE
[NOD-862] Replace calls to Tx.StoreBlock, Tx.HasBlock, Tx.FetchBlock with appropriate calls in dbaccess

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: dbaccess/block.go [hunk 1]
```diff
@@ -16,28 +15,23 @@ var (
 )
 
 // StoreBlock stores the given block in the database.
-func StoreBlock(context Context, block *util.Block) error {
+func StoreBlock(context Context, hash []byte, blockBytes []byte) error {
```
Comment (someone235) [inline dbaccess/block.go]: Please use hash as *daghash.Hash here and everywhere else in dbaccess

File: server/rpc/handle_get_block.go [hunk 2]
```diff
@@ -39,18 +38,20 @@ func handleGetBlock(s *Server, cmd interface{}, closeChan <-chan struct{}) (inte
...
 
-	var blkBytes []byte
-	err = s.cfg.DB.View(func(dbTx database.Tx) error {
-		var err error
-		blkBytes, err = dbTx.FetchBlock(hash)
-		return err
-	})
-	if err != nil {
+	blkBytes, found, err := dbaccess.FetchBlock(dbaccess.NoTx(), hash[:])
+	if err != nil || !found {
```
Comment (someone235) [inline server/rpc/handle_get_block.go]: You probably don't want to return "Block not found" for every error

File: blockdag/accept.go [hunk 3]
```diff
@@ -62,13 +62,29 @@ func (dag *BlockDAG) maybeAcceptBlock(block *util.Block, flags BehaviorFlags) er
 	// expensive connection logic. It also has some other nice properties
 	// such as making blocks that never become part of the DAG or
 	// blocks that fail to connect available for further analysis.
-	err = dag.db.Update(func(dbTx database.Tx) error {
-		err := dbStoreBlock(dbTx, block)
+	dbTx, err := dbaccess.NewTx()
```
Comment (svarogg) [inline blockdag/accept.go]: Always defer RollbackIfNotClosed right after calling NewTx()

File: blockdag/dag.go [hunk 4]
```diff
@@ -599,7 +600,7 @@ func (dag *BlockDAG) saveChangesFromBlock(block *util.Block, virtualUTXODiff *UT
 
 	// Atomically insert info into the database.
 	err := dag.db.Update(func(dbTx database.Tx) error {
-		err := dag.index.flushToDBWithTx(dbTx)
+		err := dag.index.flushToDBWithContext(dbaccess.NoTx()) // TODO: (Stas) Replace this with a tx context
```
[the remark this answers is not in the document]
Comment (svarogg) [reply blockdag/dag.go]: What Ori said ;)

File: blockdag/dag_test.go [hunk 5]
```diff
@@ -554,8 +555,13 @@ func TestNew(t *testing.T) {
 	if err != nil {
 		t.Fatalf("error creating db: %s", err)
 	}
+	err = dbaccess.Open(dbPath)
+	if err != nil {
+		t.Fatalf("error creating db: %s", err)
+	}
 	defer func() {
 		db.Close()
+		dbaccess.Close()
```
[the remark this answers is not in the document]
Comment (svarogg) [reply blockdag/dag_test.go]: Why do we even need both db and dbaccess?
Is it something temporary for the time before we finish migrating to dbaccess?

File: blockdag/dag_test.go [hunk 6]
```diff
@@ -625,13 +636,19 @@ func TestAcceptingInInit(t *testing.T) {
...
 
 	// Manually add the test block to the database
-	err = db.Update(func(dbTx database.Tx) error {
-		err := dbStoreBlock(dbTx, testBlock)
-		if err != nil {
-			return err
-		}
-		return dbStoreBlockNode(dbTx, testNode)
-	})
+	testBlockBytes, err := testBlock.Bytes()
```
Comment (svarogg) [inline blockdag/dag_test.go]: This code is very similar to the code you wrote in accept.go - consider extracting to a function

File: server/rpc/handle_get_selected_tip.go [hunk 7]
```diff
@@ -12,32 +12,27 @@ func handleGetSelectedTip(s *Server, cmd interface{}, closeChan <-chan struct{})
...
 
-	var blockBytes []byte
-	err := s.cfg.DB.View(func(dbTx database.Tx) error {
-		var err error
-		blockBytes, err = dbTx.FetchBlock(selectedTipHash)
-		return err
-	})
-	if err != nil {
+	blockBytes, found, err := dbaccess.FetchBlock(dbaccess.NoTx(), selectedTipHash[:])
+	if err != nil || !found {
```
Comment (svarogg) [inline server/rpc/handle_get_selected_tip.go]: Here too (what Ori said a comment ago)

File: blockdag/dag_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (stasatdaglabs) [reply blockdag/dag_test.go]: db is the old database. We'll remove all references to it in the cleanup ticket.

File: blockdag/test_utils.go [hunk 8]
```diff
@@ -77,22 +79,30 @@ func DAGSetup(dbName string, config Config) (*BlockDAG, func(), error) {
...
+			return nil, nil, errors.Errorf("error creating db: %s", err)
+		}
 
 		// Setup a teardown function for cleaning up. This function is
 		// returned to the caller to be invoked when it is done testing.
 		teardown = func() {
 			spawnWaitGroup.Wait()
 			spawn = realSpawn
 			config.DB.Close()
+			dbaccess.Close()
```
[the remark this answers is not in the document]
Comment (stasatdaglabs) [reply blockdag/test_utils.go]: db is the old database. We'll remove all references to it in the cleanup ticket.

**Pull request merged**